pygeoapi/process/base.py

```diff
@@ -53,6 +53,13 @@ def __init__(self, processor_def: dict, process_metadata: dict):
         self.metadata = process_metadata
         self.supports_outputs = False
 
+    def set_job_id(self, job_id: str) -> None:
+        """
+        set the job_id within the processor
+        To be implemented by derived classes where required.
+        """
```

Comment:
Can you add a description of `job_id` in the docstring? i.e.

```python
 def set_job_id(self, job_id: str) -> None:
        """
        Sets a job identifier explicitly

        To be implemented by derived classes as required
        """

        :param job_id: job identifier

        :returns: `None`

        pass
```

Reply:
@francescoingv thanks for the update.  The return type needs to be added to the docstring.  As well, the `job_id` description needs the `param prefix`, and the text of the `job_id` descripition needs to be aligned.  So:

```python
 def set_job_id(self, job_id: str) -> None:
        """
        Sets a job identifier explicitly

        To be implemented by derived classes as required
        """

        set the job_id within the processor

        To be implemented by derived classes where required.

        :param job_id: the job_id assigned to the request by the Manager.
                    The function shuold be called by the Manager upon assigning
                    the job_id. The job_id is intended to be used by derived
                    classes, e.g. to write temporary files where filenames
                    contains the string job_id.

        :returns: `None`

        pass
```

Reply:
@tomkralidis Thank you